### PR TITLE
GROOVY-7948: fix completion of static imports in groovysh

### PR DESCRIPTION
**Pull request by heenenee**
Fixed completion of static imports; updated unit tests; did some refactorings in ImportsSyntaxCompletor; added comment explaining how imports are stored in Groovysh.  Also did some manual testing of completion functionality using groovysh.bat in a cleanly built binary distribution.

### CODE REVIEW

**Comment by blackdrag** [comment]
This change is probably to be considered as a breaking change. If John is ok with this I am ok with having this on master for 2.5.0

**Comment by jwagenleitner** [comment]
Thanks for the PR.

I haven't gone through it in detail, but wonder if we didn't use the diamond operator on `cachedImports` and retained the method signature of `collectImportedSymbols` if this fix would be eligible for 2_4_X?

**Comment by heenenee** [comment]
I could get rid of the diamond operator and preserve the method signatures,
and incorporate whatever other feedback if it helps get the fix released
sooner.

**Comment by jwagenleitner** [comment]
I forgot to mention that while the changes to the test (removing `import`) seems to be a significant change, I think this was a problem with the tests not matching the actual behavior of Groovysh since it stores imports with the `import` as mentioned by the added comment.

In addition to fixing completion of static imports, I also found this fixes the completion of star imports.  Prior to this fix `import java.text.*` and typing `new SimpleDa<tab>` would not return `SimpleDateFormat`, after this fix it does.

File: subprojects/groovy-groovysh/src/main/groovy/org/codehaus/groovy/tools/shell/completion/ImportsSyntaxCompletor.groovy [hunk 1]
```diff
@@ -87,51 +78,49 @@ class ImportsSyntaxCompletor implements IdentifierCompletor {
...
      * finds matching imported classes or static methods
-     * @param prefix
-     * @param importSpec
-     * @param matches
-     * @return
+     * @param importSpec an import statement without the leading 'import ' or trailing semicolon
+     * @return all names matching the importSpec
      */
-    void collectImportedSymbols(final String importSpec, final Collection<String> matches) {
+    SortedSet<String> collectImportedSymbols(final String importSpec) {
```
Comment (blackdrag) [inline subprojects/groovy-groovysh/src/main/groovy/org/codehaus/groovy/tools/shell/completion/ImportsSyntaxCompletor.groovy]: we decided to be strict about such things in a bugfix release and 2.4.8 is one. So yeah, unless the method signature is made compatible no 2.4.x

**Comment by heenenee** [comment]
I removed the diamond operator and preserved the method signature and parameter names of `collectImportedSymbols` to make the fix available to 2.4.x.

**Comment by jwagenleitner** [comment]
PR merged for 2.4.x and 2.5, thanks!

**Pull request closed without merging**